### PR TITLE
Var constructors: new_param, new_obs, new_calc, new_value

### PR DESCRIPTION
**Pull request by jobrachem**
For a while now, I have been a little unsatisfied with two things:

1. Incsonsistent UI: `Var` class vs. `param` and `obs` functions. See also #130 
2. The status of the `Data` (or, in #170 : `Value`) node vs. `Var`: Which should you use?

I think there is a satisfying solution:

1. Implement constructors for `Var` that fullfill the tasks above. 
2. Guide users to towards using `Var.new_<constructor>` as a default.

If I recall correctly, using constructors has been proposed before by @wiep.

This PR implements four constructors. 

### Var.new_param

Supersedes `lsl.param()`

```python
class Var:

    @classmethod
    def new_param(
        cls, value: Any, distribution: Dist | None = None, name: str = ""
    ) -> Self:
        ...
```

### Var.new_obs

Supersedes `lsl.obs()`

```python
class Var:
    @classmethod
    def new_obs(
        cls, value: Any, distribution: Dist | None = None, name: str = ""
    ) -> Self:
        ...
```

### Var.new_calc

New. Having this constructor available makes the interface more consistent. Users can simply start typing `Var.new_` and see the available options via auto-completion. It also reduces errors from confusion about nesting `Calc` objects in `Var` constructors.

```python
class Var:
    @classmethod
    def new_calc(
        cls,
        function: Callable[..., Any],
        *inputs: Any,
        name: str = "",
        _needs_seed: bool = False,
        update_on_init: bool = True,
        **kwinputs: Any,
    ) -> Self:
        ...
```

### Var.new_const

New. Like with `Var.new_calc`, having this constructor makes the interface more consistent and supports a good habit.

```python
class Var:
    @classmethod
    def new_const(cls, value: Any, name: str = "") -> Self:
        ...
```

## Todo

This is a draft PR for a first discussion. Even if it remains unchanged, documentation has to be added if it ends up being merged.

- [x] Add documentation
- [x] Add deprecation warnings to `param` and `obs`
- [ ] (Maybe) add a warning to using the default `Var` constructor

### CODE REVIEW

**Comment by jobrachem** [comment]
Discussion notes

- The three constructors `.new_param`, `new_obs` and `new_calc` appear to be unproblematic.
- There's doubt about the need for and naming of `.new_const`. The naming could imply that it is immutable, which it is not. It is possible to change the name.
- We could go forward with this constructor, too, but add a disclaimer about it being experimental.

What is the purpose of `.new_const`?
- It is equivalent to using the default `lsl.Var` without specifying a distribution.
- So it holds a value that is neither observed nor a parameter. In most cases it is not crucial to explicitly make such a value immutable.
- It should encourage the good habit of using the `.new_` constructors.


Maybe `.new_const` just needs a better name. Maybe `lsl.Var.new_value`? That would provide a nice parallel to how `lsl.Var.new_calc` wraps a `lsl.Calc`.

**Comment by jobrachem** [comment]
@GianmarcoCallegher gentle reminder :)

**Comment by jobrachem** [comment]
@GianmarcoCallegher As we just discussed, the test action is currently failing due to timeout. What is surprising is that I did not change any code between the previous test run and the current test run. I just re-ran the action. Locally, the tests run in <150s. This is weird to me.

**Comment by jobrachem** [comment]
See https://github.com/liesel-devs/liesel/issues/213 regarding the tests timing out.

**Comment by jobrachem** [comment]
@GianmarcoCallegher the testing issue is solved for the moment, and the PR can be reviewed :)

**Review by GianmarcoCallegher — COMMENTED** [review]
All clear. Just a little comment

File: liesel/model/model.py [hunk 1]
```diff
@@ -963,8 +963,13 @@ class Model:
...
     .. tip::
-        While you can create a model directly, it is usually more convenient to use a
-        :class:`.GraphBuilder` to construct the model.
+        If you have an existing model and want to make changes to it, you can use the
+        :meth:`.Model.pop_nodes_and_vars` method to release the nodes and variables
+        of the model. You can then make changes to them, for example i.e. changing the
+        distribution of a variable or the inputs of a calculation. Afterwards, you
+        initialize a *new* model with your changed variables.
+        If you simply want to change the value of a variable, it is not necessary to
+        call :meth:`~.Model.pop_nodes_and_vars`.
```
[the remark this answers is not in the document]
Comment (jobrachem) [reply liesel/model/model.py:972]: Good point, will do that.

**Pull request merged**